shared_model/interfaces/iroha_internal/query_response_factory.hpp

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_QUERY_RESPONSE_FACTORY_HPP
+#define IROHA_QUERY_RESPONSE_FACTORY_HPP
+
+#include <memory>
+
+#include "interfaces/query_responses/block_query_response.hpp"
+#include "interfaces/query_responses/query_response.hpp"
+
+namespace shared_model {
+  namespace interface {
+
+    /**
+     * Factory for building query responses
+     */
+    class QueryResponseFactory {
+     public:
+      virtual ~QueryResponseFactory() = default;
+
+      /**
+       * Create response for account asset query
+       * @param assets to be inserted into the response
+       * @return account asset response
+       */
+      virtual std::unique_ptr<AccountAssetResponse> createAccountAssetResponse(
```

Comment:
AccountAssetResponse -> QueryResponse, same for other methods, other than block query response.
Query hash also has to be passed as arguments.